src/main/kotlin/test/ktortemplate/core/service/CarServiceImpl.kt

```diff
@@ -18,17 +17,21 @@ class CarServiceImpl : KoinComponent, CarService {
     private val partRepository: PartRepository by inject()
     private val dbc: DatabaseConnection by inject()
 
-    override suspend fun count(pageRequest: PageRequest): Int = this.carRepository.count(pageRequest)
+    override suspend fun count(pageRequest: PageRequest): Int {
```

Comment:
This illustrates that the service is responsible for setting the desired transactional semantic. This is super relevant for when we have code that does several database operations, like in the  `registerPartReplacement` method. 

In the old version of the code, it would work event if the service didn't call the `dbc.query` method (starting the parent transaction), however it would not rollback correctly if some intermediate operation failed. As each call to a repo would be its own transaction and therefore commit individually, breaking the atomicity of the operation.

Reply:
Good work @felix19350 this is a major improvement for us in terms of having the database code doing what we really want instead of relying on repositories to start their own transaction.

Reply:
👍  since these operations are mainly crud it makes senses to delegate the transaction responsibility to the consumer of the repository.